packages/react-native/converter.ios.ts

```diff
@@ -0,0 +1,116 @@
+  const nativeArguments = [];
+
+  for (let i = 0; i < argumentTypes.length; i++) {
+    const argType = argumentTypes[i];
+    const data = args[i];
+    switch (argType) {
+      case RNObjcSerialisableType.array: {
+        nativeArguments.push(Utils.ios.collections.jsArrayToNSArray(data));
+        break;
+      }
+      case RNObjcSerialisableType.nonnullArray: {
+        if (!data) throw new Error(`Argument at index ${i} expects a nonnull Array value`);
+        nativeArguments.push(Utils.ios.collections.jsArrayToNSArray(data));
+        break;
+      }
+      case RNObjcSerialisableType.object: {
+        nativeArguments.push(Utils.dataSerialize(data));
+        break;
+      }
+      case RNObjcSerialisableType.nonnullObject: {
+        if (!data) throw new Error(`Argument at index ${i} expects a nonnull Object value`);
+        nativeArguments.push(Utils.dataSerialize(data));
+        break;
+      }
+      case RNObjcSerialisableType.boolean:
+      case RNObjcSerialisableType.string:
+      case RNObjcSerialisableType.other:
+      case RNObjcSerialisableType.number:
+        nativeArguments.push(data);
+        break;
+      case RNObjcSerialisableType.nonnullBoolean:
+        if (Utils.isBoolean(data)) throw new Error(`Expected a boolean but got ${data}`);
+        nativeArguments.push(data);
+        break;
+      case RNObjcSerialisableType.nonnullString:
+        if (Utils.isString(data)) throw new Error(`Expected a string but got ${data}`);
+        nativeArguments.push(data);
+        break;
+      case RNObjcSerialisableType.nonnullNumber:
+        if (!Utils.isNumber(data)) throw new Error(`Expected a number but got ${data}`);
+        nativeArguments.push(data);
+        break;
+      case RNObjcSerialisableType.RCTResponseSenderBlock: {
+        nativeArguments.push(
+          !data
+            ? undefined
+            : (value: unknown[]) => {
+                data(...(toJSValue(value) as unknown[]));
+              }
+        );
+        break;
+      }
+      case RNObjcSerialisableType.RCTResponseErrorBlock: {
+        nativeArguments.push(
+          !data
+            ? undefined
+            : (value: NSError) => {
+                data(toJSValue(value.userInfo));
+              }
+        );
+        break;
+      }
+      case RNObjcSerialisableType.RCTPromiseResolveBlock: {
+        nativeArguments.push((value: unknown) => {
+          resolve(toJSValue(value));
+        });
+        break;
+      }
+      case RNObjcSerialisableType.RCTPromiseRejectBlock: {
+        nativeArguments.push((reason: unknown) => {
+          reject(toJSValue(reason));
+        });
+        break;
+      }
+      default:
+        break;
+    }
+  }
+  return nativeArguments;
+}
+
+export function promisify(module: RCTBridgeModule, methodName: string, argumentTypes: RNObjcSerialisableType[], args) {
```

Comment:
```suggestion
export function promisify(module: RCTBridgeModule, methodName: string, argumentTypes: RNObjcSerialisableType[], args: unknown[]): Promise<unknown> {
```
We might be able to return better than `Promise<unknown>` using some generics based on `module` and `methodName`, but I'll have to come back and have a think about it. It would be possible if all the modules were registered onto a common interface (and any new modules augmented that interface). But as it might not be possible, I'm not going to call it a dealbreaker.

Reply:
We probably don't need it as the end users will be using the existing JS layer written by the library creator and not NativeModules directly. Even in React Native, I haven't found any types directly on NativeModules.MyModule.function unless explicitly added by the library creator. Hope that makes sense.